index-import.cpp

```diff
@@ -59,18 +59,33 @@ static cl::opt<bool> UndoRulesSwiftRenames(
 
 struct Remapper {
 public:
+  explicit Remapper() { llvm::sys::fs::current_path(this->pwd); }
+
   std::string remap(const llvm::StringRef input) const {
+    if (input.empty()) {
+      return input.str();
+    }
+
     std::string input_str = input.str();
+    std::optional<StringRef> output_str = std::nullopt;
     for (const auto &remap : this->_remaps) {
       const auto &pattern = remap.first;
       const auto &replacement = remap.second;
       if (re2::RE2::Replace(&input_str, *pattern, replacement)) {
-        return path::remove_leading_dotslash(StringRef(input_str)).str();
+        output_str = path::remove_leading_dotslash(StringRef(input_str));
+        break;
       }
     }
 
-    // No patterns matched, return the input unaltered.
-    return path::remove_leading_dotslash(input).str();
+    StringRef out = output_str.value_or(path::remove_leading_dotslash(input));
+    if (path::is_absolute(out)) {
+      return out.str();
+    }
+
+    SmallString<128> abs(out);
+    llvm::sys::fs::make_absolute(this->pwd, abs);
+    llvm::outs() << "resulting path is " << abs << "\n";
```

Comment:
maybe a warning when this branch is run

Reply:
yea I guess we should, but the maybe we should have a flag to ignore the warnings or something? because I imagine that would be pretty noisy

Reply:
yeah something like `--quiet` or `-q` which would just return exit code should be enough.

Reply:
What's the point of this output here? I agree if we need something, we should allow to make it quiet. Do we expect not to see it during an Xcode-Bazel (i.e. rules_xcodeproj) integration, so seeing it is a notice to fix something, or will we see it and we really can't do anything about it?

Reply:
Theoretically this means you're possibly missing a remap and that your index quality could be affected. At least that's the assumption until we figure out why @yongjincho92 found some paths hitting this

Reply:
> Do we expect not to see it during an Xcode-Bazel (i.e. rules_xcodeproj) integration, so seeing it is a notice to fix something, or will we see it and we really can't do anything about it?

Former is the case.

Reply:
i'm just going to leave it out for now until we care enough to add a flag. if you haven't noticed any indexing issues maybe it doesn't matter.